charts/veritable-cloudagent/templates/deployment.yaml

```diff
@@ -0,0 +1,301 @@
+          {{- if .Values.diagnosticMode.enabled }}
+          args: {{- include "common.tplvalues.render" (dict "value" .Values.diagnosticMode.args "context" $) | nindent 12 }}
+          {{- else if .Values.args }}
+          args: {{- include "common.tplvalues.render" (dict "value" .Values.args "context" $) | nindent 12 }}
+          {{- end }}
+          env:
+            - name: VERITABLE_DEBUG
+              value: {{ ternary "true" "false" (or .Values.image.debug .Values.diagnosticMode.enabled) | quote }}
+            - name: AFJ_REST_INBOUND_TRANSPORTS
+              value: "http {{ .Values.containerPorts.http }} ws {{ .Values.containerPorts.ws }}"
+            - name: AFJ_REST_LABEL
+              value: {{ .Values.label | quote }}
+            - name: AFJ_REST_WALLET_ID
+              value: {{ .Values.walletId | quote }}
+            - name: AFJ_REST_WALLET_KEY
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "veritable-cloudagent.walletSecretName" . }}
+                  key: {{ include "veritable-cloudagent.walletSecretKey" . }}
+            - name: AFJ_REST_STORAGE_TYPE
+              value: {{ .Values.storageType | quote }}
+            - name: AFJ_REST_ENDPOINT
+              value: {{ .Values.endpoint | quote }}
+            - name: AFJ_REST_LOG_LEVEL
+              value: {{ .Values.logLevel | quote }}
+            - name: AFJ_REST_USE_DID_SOV_PREFIX_WHERE_ALLOWED
+              value: {{ .Values.useDidSovPrefixWhereAllowed | quote }}
+            - name: AFJ_REST_USE_DID_KEY_IN_PROTOCOLS
+              value: {{ .Values.useDidKeyInProtocols | quote }}
+            - name: AFJ_REST_OUTBOUND_TRANSPORT
+              value: {{ .Values.outboundTransport | quote }}
+            - name: AFJ_REST_AUTO_ACCEPT_CONNECTIONS
+              value: {{ .Values.autoAcceptConnections | quote }}
+            - name: AFJ_REST_AUTO_ACCEPT_CREDENTIALS
+              value: {{ .Values.autoAcceptCredentials | quote }}
+            - name: AFJ_REST_AUTO_ACCEPT_MEDIATION_REQUESTS
+              value: {{ .Values.autoAcceptMediationRequests | quote }}
+            - name: AFJ_REST_AUTO_ACCEPT_PROOFS
+              value: {{ .Values.autoAcceptProofs | quote }}
+            - name: AFJ_REST_AUTO_UPDATE_STORAGE_ON_STARTUP
+              value: {{ .Values.autoUpdateStorageOnStartup | quote }}
+            - name: AFJ_REST_BACKUP_BEFORE_STORAGE_UPDATE
+              value: {{ .Values.backupBeforeStorageUpdate | quote }}
+            {{- if .Values.connectionImageUrl }}
+            - name: AFJ_REST_CONNECTION_IMAGE_URL
+              value: {{ .Values.connectionImageUrl | quote }}
+            {{- end }}
+            {{- if .Values.webhookUrl }}
+            - name: AFJ_REST_WEBHOOK_URL
+              value: {{ .Values.webhookUrl | quote }}
+            {{- end }}
+            - name: AFJ_REST_ADMIN_PORT
+              value: {{ .Values.adminPort | quote }}
+            - name: AFJ_REST_IPFS_ORIGIN
```

Comment:
Given we're always deploying an ipfs locally is there any reason for this? It would seem more sensible to either hardcode this to be on the same host or to make deploying the extra container optional and then refer to an external ipfs

Reply:
Resolved in latest commits, made ipfs container conditional on `ipfs.enabled` being true.  Changes the ipfsOrigin in `values.yaml` to reflect that it is an external origin and updated the deployment to use the localhost container if `ipfs.enabled` is true.